### DISCUSSION TITLE
Inquiry about using Duckdb to implement the TPC-DS and TPC-DI

### DISCUSSION THREAD

**Discussion by woshimajintao** [duckdb/duckdb#9221] [Q&A]
I am  a beginner of Duckdb.

We want to use the Duckdb to Implement the TPC-DS and TPC-DI benchmark.

https://www.tpc.org/tpcds/
https://www.tpc.org/tpcdi/

I would like to know is there anyone have tried to do it and written some documents and tutorials? Or is there anyone could give me some advices about it?

I hope to some reply.If you have some advices,you can contact me mars19990123@gamil.com

**Comment by Tishj** [comment]
I think we provide tpcds as an extension already

https://duckdb.org/docs/extensions/tpcds

**Reply by woshimajintao** [reply]
Thank you!I wonder how can we implement the TPC-DI benchmark using Duckdb？

**Reply by Tishj** [reply]
I'd say have a look at the code for the tpcds extension and copy+adapt it for TPC-DI ?
<https://github.com/duckdb/duckdb/tree/main/extension/tpcds>

**Comment by l1t1** [comment]
TPC-DI is a data integration benchmark.  duckdb is an OLAP database

**Reply by woshimajintao** [reply]
So can't I use the Duckdb to implement the TPC-DI benchmark?

**Reply by djouallah** [reply]
actually you can, I suspect DuckDB will end up as a great tool for Data integration
